base_layer/core/src/base_node/rpc/query_service.rs

```diff
@@ -269,7 +271,12 @@ impl<B: BlockchainBackend + 'static> Service<B> {
 
             fetched_utxos += 1;
 
-            if current_header.height >= end_header.height || fetched_utxos >= request.limit {
+            if current_header.height >= tip_header.header().height {
+                next_header_to_request = vec![];
```

Comment:
```suggestion
                next_header_to_request = None;
```

Reply:
its an vec not an option type, I need to do vec<vec<u8>> if I want to do that